hermes-plugin/src/main/groovy/com/atlasgenetics/hermes/response/HermesResponseWrapper.groovy

```diff
@@ -0,0 +1,36 @@
+package com.atlasgenetics.hermes.response
+
+import com.atlasgenetics.hermes.utils.HttpStatusUtils
+import groovy.transform.CompileStatic
+import groovyx.net.http.HttpResponseDecorator
+import org.apache.http.Header
+
+class HermesResponseWrapper {
+    int statusCode
+    Map<String, Object> headers
+    def payload
+    Long failedMessageId
+
+    void setHeaders(Header[] allHeaders) {
+        headers = [:]
```

Comment:
If you want to groovify this you can do `allHeaders.collectEntries { [it.name, it.value] }`. `collectEntries` will return an empty map should the collection of headers be empty.